File: Go/Sqrt(x).go
package main

func main() {

}

func mySqrt(x int) int {
	if x == 0 || x == 1 {
		return x
	}

	i, result := 1, 1

	for result <= x {
		i++
		result = i * i
	}

	return i - 1
}

File: Javascript/Sqrt(x).js
/**
 * @param {number} x
 * @return {number}
 */
var mySqrt = function (x) {
  if (x == 0 || x == 1) {
    return x;
  }

  let [i, result] = [1, 1];

  while (result <= x) {
    i++;
    result = i * i;
  }

  return i - 1;
};
